src/main/java/com/dreammaster/gthandler/multiAirFilter/GT_MetaTileEntity_AirFilterBase.java

```diff
@@ -112,109 +141,151 @@ public boolean isFacingValid(byte aFacing) {
         return aFacing > 1;
     }
 
+    public int getPollutionCleaningRatePerTick(float turbineEff, float multiEff, boolean isRateBoosted){return getPollutionCleaningRatePerSecond(turbineEff, multiEff, isRateBoosted) / 20;
+    }
+
+    public int getPollutionCleaningRatePerSecond(float turbineEff, float multiEff, boolean isRateBoosted){
+        long tVoltage = getMaxInputVoltage();
+        byte tTier = (byte) max(1, GT_Utility.getTier(tVoltage));
+        int pollutionPerSecond = 0;
+        for (GT_MetaTileEntity_Hatch_Muffler tHatch : mMufflerHatches) {
+            if (isValidMetaTileEntity(tHatch)) {
+                //applying scaling factor
+                pollutionPerSecond += (int) Math.pow(MainRegistry.CoreConfig.scalingFactor, min(tTier, tHatch.mTier));
+            }
+        }
+        //apply the boost
+        if (isRateBoosted){
+            pollutionPerSecond = (int) (pollutionPerSecond * MainRegistry.CoreConfig.boostPerAbsorptionFilter);
+        }
+        //apply the rest of the coefs
+        pollutionPerSecond = (int) (pollutionPerSecond * turbineEff * multiEff * getBonusByTier() * MainRegistry.CoreConfig.globalMultiplicator);
+        return pollutionPerSecond;
+    }
+
     @Override
     public boolean checkRecipe(ItemStack aStack){
         mEfficiencyIncrease = 10000;
         mEfficiency = 10000 - (getIdealStatus() - getRepairStatus()) * 1000;
-        mPollutionReductionWholeCycle = 0;
-        mMaxProgresstime = 200;
-        mEUt=-32;
+        // check pollution for next cycle:
+        hasPollution = getTotalPollution() >= MainRegistry.CoreConfig.pollutionThresholdAirFilter;
+        mMaxProgresstime = getRecipe().mDuration;
+        mEUt=-getRecipe().mEUt;
         if(!hasPollution) {
```

Comment:
Where goes the overclock handling? Also, isn't it supposed to consume less EU/t when it doesn't have pollution?

Reply:
It doesn’t overclock iirc